cfme/infrastructure/virtual_machines.py

```diff
@@ -1670,13 +1696,17 @@ class TemplatesAll(CFMENavigateStep):
     prerequisite = NavigateToSibling('All')
 
     def step(self, *args, **kwargs):
-        if 'filter_folder' not in kwargs:
-            self.view.sidebar.templates.tree.click_path('All Templates')
-        elif 'filter_folder' in kwargs and 'filter_name' in kwargs:
-            self.view.sidebar.templates.tree.click_path('All Templates', kwargs['filter_folder'],
-                                                        kwargs['filter_name'])
+        if self.obj.parent is self.obj.appliance:
+            if 'filter_folder' not in kwargs:
+                self.view.sidebar.templates.tree.click_path('All Templates')
+            elif 'filter_folder' in kwargs and 'filter_name' in kwargs:
+                self.view.sidebar.templates.tree.click_path('All Templates',
+                                                            kwargs['filter_folder'],
+                                                            kwargs['filter_name'])
+            else:
+                raise DestinationNotFound("the destination isn't found")
         else:
-            raise DestinationNotFound("the destination isn't found")
+            self.prerequisite_view.entities.summary("Relationships").click_at("Templates")
```

Comment:
Have you tested this navigation path? The prerequisite view here is InfraTemplateCollection's 'All' destination, which won't have a relationships summary table.

Reply:
What I was trying to point out in previous comments is that this navigation destination already exists as `ProviderTemplates` for the provider instance. `InfraTemplateCollection.all` is using it.

https://github.com/ManageIQ/integration_tests/blob/master/cfme/infrastructure/provider/__init__.py#L446

The navigation instance and destination need to be picked to be able to use this, and I think that can be done in Comparable.

Reply:
I am looking at this now and testing, but I believe I handled this before I get to TemplatesAll. I should have deleted the extra code meant for providers , but I don't think I'd ever get here with a provider, as I changed InfraTemplateCollection to pass NAV_STRING and COMPARE_VIEW based upon having a provider filter or not. I am going to go though and test everything just to make sure and will fix anything I find. Thanks.

Reply:
Yeah. I meant to remove the code I added to TemplatesAll. It has been removed now, but I have a bug in the navigation to work out. I have removed the logic for determining provider vs. appliance out of navigation step and view classes. Should have the bug fixed shortly. I am testing thoroughly.